### PR TITLE
Make newValidateSpec and newSanitizeSpec public

### PR DESCRIPTION
**Pull request by harikt**
As the Filter still cannot be used directly with Aura.Input without a common interface :-/ .

### CODE REVIEW

**Comment by harikt** [comment]
Not sure why the test fails only for 5.3 and 5.4  . Evaluated over http://3v4l.org/q0q56 . Seems a travis issue ?

File: src/FilterFactory.php [hunk 1]
```diff
@@ -67,12 +67,12 @@ public function newValueFilter()
         );
     }
 
-    protected function newValidateSpec()
+    public function newValidateSpec()
```
Comment (Ocramius) [inline src/FilterFactory.php:70]: The naming of this API really doesn't suggest that it should be `public` :-\

Comment (harikt) [inline src/FilterFactory.php:70]: @Ocramius not sure I get you. Could you please expand ?

**Comment by koriym** [comment]
> Not sure why the test fails only for 5.3 and 5.4 

Tests fails because class name resolution is not supported 5.3 and 5.4.
It failed since https://github.com/auraphp/Aura.Filter/commit/7302ed4d7099e864f64315e3dda9478f4b1d17c4#diff-b84f28b99c9ec288cc3e2f51ba0bf698R15

**Comment by harikt** [comment]
Good catch @koriym . Thanks!

**Comment by pmjones** [comment]
In theory, the *Spec classes are only for operations internal to the Filter class, and not for public consumption. There's no point to making them public, at least not yet. I retain the right to revise my opinion at a later time. ;-)

**Comment by harikt** [comment]
It was added to make convenience with Aura.Input filter without Di . 

But I understood version 2 of Filter will not work with Aura.Input and need some work. So did learned we don't need Aura.Input, but only a validation library for form, so the input all are rendered same as html not rendered via a helper like Aura.Html when using other templating systems.

I did something over https://github.com/harikt/conduit-skelton/blob/master/src/Form/ContactForm.php 
The more configuration for Aura.View to load the template is why I did looked for a different way, though I like the way Aura.Input and Aura.Html works .

**Pull request closed without merging**